starboard/shared/starboard/player/decoded_audio_internal.cc

```diff
@@ -161,17 +161,26 @@ void DecodedAudio::AdjustForDiscardedDurations(
   SB_DCHECK(discarded_duration_from_back >= 0);
   SB_DCHECK(storage_type() == kSbMediaAudioFrameStorageTypeInterleaved);
 
-  const auto bytes_per_frame = GetBytesPerSample(sample_type()) * channels_;
-  auto discarded_frames_from_front =
-      AudioDurationToFrames(discarded_duration_from_front, sample_rate);
+  if (discarded_duration_from_front == 0 && discarded_duration_from_back == 0) {
+    return;
+  }
 
-  discarded_frames_from_front = std::min(discarded_frames_from_front, frames());
+  const auto bytes_per_frame = GetBytesPerSample(sample_type()) * channels_;
+  int current_frames = frames();
+  int discarded_frames_from_front =
+      (discarded_duration_from_front >=
+       AudioFramesToDuration(current_frames, sample_rate))
+          ? current_frames
+          : AudioDurationToFrames(discarded_duration_from_front, sample_rate);
   offset_in_bytes_ += bytes_per_frame * discarded_frames_from_front;
   size_in_bytes_ -= bytes_per_frame * discarded_frames_from_front;
 
-  auto discarded_frames_from_back =
-      AudioDurationToFrames(discarded_duration_from_back, sample_rate);
-  discarded_frames_from_back = std::min(discarded_frames_from_back, frames());
```

Comment:
Would the overflow be resolved by replacing this line with the following?
discarded_frames_from_back = std::min(discarded_frames_from_back, frames() - discarded_frames_from_front);

Reply:
The overflow is in AudioDurationToFrames() for `duration * samples_per_second`. If discarded_duration_from_back is too large, the returned value of AudioDurationToFrames() would be wrong.